src/sentry/api/endpoints/organization_events_spans_performance.py

```diff
@@ -251,12 +252,33 @@ def data_fn(offset: int, limit: int) -> Any:
         with self.handle_query_errors():
             return self.paginate(
                 request,
-                paginator=GenericOffsetPaginator(data_fn=data_fn),
+                paginator=SpanExamplesPaginator(data_fn=data_fn),
                 default_per_page=3,
                 max_per_page=10,
             )
 
 
+class SpanExamplesPaginator:
+    def __init__(self, data_fn: Callable[[int, int], Any]):
+        self.data_fn = data_fn
+
+    def get_result(self, limit: int, cursor: Optional[Cursor] = None) -> CursorResult:
+        assert limit > 0
```

Comment:
Should we throw an error or log something here if the limit is incorrect instead of a bare assert?

Reply:
This was taken from `GenericOffsetPaginator` see https://github.com/getsentry/sentry/blob/bdd7c593d40bcf6897d9d6f69f93ed75f4ef8721/src/sentry/api/paginator.py#L480

Can definitely add a better error message here though.